src/SIL.XForge.Scripture/Services/SFProjectService.cs

```diff
@@ -502,13 +511,35 @@ public async Task<string> GetLinkSharingKeyAsync(string curUserId, string projec
                     {
                         Key = key,
                         ProjectRole = role,
-                        ExpirationTime = null
+                        ExpirationTime = (shareLinkType == ShareLinkType.Recipient ? expTime : null),
+                        ShareLinkType = shareLinkType
```

Comment:
You may want to add a comma at the end of `ShareLinkType = shareLinkType` (i.e. change to `ShareLinkType = shareLinkType,`), so that if a property is added to ShareKey in future, it will not require this line to be modified (and so make that PR cleaner).